### PR TITLE
STM32 USB V2 Updates STML4

### PR DESCRIPTION
**Pull request by psidhu**
Here's the updates we spoke about in https://github.com/zephyrproject-rtos/zephyr/pull/5163

There's one caveat: I can't seem to use LL_PWR_EnableVddUSB() since the .h isn't being included. I've commented this out and instead used HAL_PWREx_EnableVddUSB(), unless you know of a clean method of including the stm32l4xx_ll_pwr.h.

### CODE REVIEW

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -309,6 +317,20 @@ int usb_dc_attach(void)
 		return ret;
 	}
 
+	/* Enable VDDUSB if available */
```
Comment (ydamigos) [inline drivers/usb/device/usb_dc_stm32.c]: We should mention in the comment that we need the following code for stm32l4 series and why we need it (a part from DM00310109 section 5.1.3 should do it).

File: drivers/usb/device/usb_dc_stm32.c [hunk 2]
```diff
@@ -309,6 +317,20 @@ int usb_dc_attach(void)
...
+#ifdef PWR_CR2_PVME1
+	if (LL_APB1_GRP1_IsEnabledClock(LL_APB1_GRP1_PERIPH_PWR)) {
+		/* LL_PWR_EnableVddUSB(); */
```
Comment (ydamigos) [inline drivers/usb/device/usb_dc_stm32.c]: I would add the following in `arch/arm/soc/st_stm32/stm32l4/soc.h` to include `stm32l4xx_ll_pwr.h`:
```
#ifdef CONFIG_USB
/* Needed to remove USB OTG peripheral electrical isolation */
#include <stm32l4xx_ll_pwr.h>
#endif
```

**Comment by psidhu** [comment]
@ydamigos done. please also add your sign-offs to these commits as you see fit.

**Comment by ydamigos** [comment]
@psidhu Thanks for the changes. I merged them.
Which board, soc, sample did you use to test usb on stm32l4? I would like to add the information to my PR's message.

**Comment by psidhu** [comment]
@ydamigos I used the nucleo_l4r5zi board which has the STM32L4R5ZI soc. I'm in the middle of getting this upstreamed: https://github.com/zephyrproject-rtos/zephyr/pull/5541

I tested ``subsys/usb/cdc_acm``, then ``cdc_eem`` using the following two examples: ``net/throughput_server`` and ``net/http_server``.

**Comment by ydamigos** [comment]
Thanks for the info, I will update PR's message.

**Pull request merged**